### PR TITLE
OCP-21364 - Create pod with both sriov-cni plugin and macvlan

### PR DESCRIPTION
**Pull request by mmirecki**

### CODE REVIEW

File: test/conformance/tests/sriov_operator.go [hunk 1]
```diff
@@ -913,9 +913,72 @@ var _ = Describe("[sriov] operator", func() {
...
 			})
 		})
+
+		Context("SRIOV and macvlan", func() {
+			// 25834
+			It("Should be able to create a pod with both sriov and macvlan interfaces", func() {
+
+				resourceName := "sriovnic"
+				sriovNetworkName := "sriovnetwork"
+				testNode := sriovInfos.Nodes[0]
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:923]: should we check the len of the nodes or we already do it in the before each?

File: test/conformance/tests/sriov_operator.go [hunk 2]
```diff
@@ -913,9 +913,72 @@ var _ = Describe("[sriov] operator", func() {
...
+				Expect(err).ToNot(HaveOccurred())
+				Eventually(func() error {
+					netAttDef := &netattdefv1.NetworkAttachmentDefinition{}
+					return clients.Get(context.Background(), runtimeclient.ObjectKey{Name: "sriovnetwork", Namespace: namespaces.Test}, netAttDef)
+				}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+				macvlanNadName := "macvlan-nad"
+				nodeNicName := sriovInfos.States[testNode].Status.Interfaces[0].Name
+				macvlanNad := network.CreateMacvlanNetworkAttachmentDefinition(macvlanNadName, namespaces.Test, nodeNicName)
+				defer clients.Delete(context.Background(), &macvlanNad)
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:941]: please use the `AfterEach` in this context section

File: test/conformance/tests/sriov_operator.go [hunk 3]
```diff
@@ -913,9 +913,72 @@ var _ = Describe("[sriov] operator", func() {
...
+				stdout, _, err := pod.ExecCommand(clients, createdPod, "ethtool", "-i", "net1")
+				Expect(err).ToNot(HaveOccurred())
+				sriovVfDriver := getDriver(stdout)
+				Expect(cluster.IsDriverSupported(sriovVfDriver[:len(sriovVfDriver)-2])).To(BeTrue())
+
+				stdout, _, err = pod.ExecCommand(clients, createdPod, "ethtool", "-i", "net2")
+				macvlanDriver := getDriver(stdout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(macvlanDriver).To(Equal("macvlan"))
+
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:964]: nit: remove this empty line please

File: test/conformance/tests/sriov_operator.go [hunk 4]
```diff
@@ -913,9 +913,72 @@ var _ = Describe("[sriov] operator", func() {
...
+			})
+		})
 	})
 })
 
+func getDriver(ethtoolstdout string) string {
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:970]: can you move this to the utils folder and make the function return `(string,error)`

File: test/conformance/tests/sriov_operator.go (hunk 1 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:923]: Checked here:
https://github.com/openshift/sriov-network-operator/blob/master/test/util/cluster/cluster.go#L56

File: test/conformance/tests/sriov_operator.go (hunk 2 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:941]: Done

File: test/conformance/tests/sriov_operator.go (hunk 3 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:964]: Done

File: test/conformance/tests/sriov_operator.go (hunk 4 above)
Comment (mmirecki) [reply test/conformance/tests/sriov_operator.go:970]: Done

File: test/conformance/tests/sriov_operator.go [hunk 5]
```diff
@@ -938,6 +938,66 @@ var _ = Describe("[sriov] operator", func() {
...
+					return clients.Get(context.Background(), runtimeclient.ObjectKey{Name: macvlanNadName, Namespace: namespaces.Test}, netAttDef)
+				}, 10*time.Second, 1*time.Second).ShouldNot(HaveOccurred())
+
+				createdPod := createTestPod(testNode, []string{sriovNetworkName, macvlanNadName})
+
+				nics, err := network.GetNicsByPrefix(createdPod, "net")
+				Expect(err).ToNot(HaveOccurred())
+				Expect(len(nics)).To(Equal(2), "Pod should have two multus nics.")
+
+				stdout, _, err := pod.ExecCommand(clients, createdPod, "ethtool", "-i", "net1")
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:988]: please put this into an Eventually like there

https://github.com/openshift/sriov-network-operator/blob/master/test/conformance/tests/sriov_operator.go#L875

some if the times the output from the ethtool connects are empty

File: test/conformance/tests/sriov_operator.go [hunk 6]
```diff
@@ -938,6 +938,66 @@ var _ = Describe("[sriov] operator", func() {
...
+				Expect(err).ToNot(HaveOccurred())
+				sriovVfDriver, err := cluster.GetNetworkDriverFromEthtool(stdout)
+				Expect(err).ToNot(HaveOccurred())
+				Expect(cluster.IsDriverSupported(sriovVfDriver[:len(sriovVfDriver)-2])).To(BeTrue())
+
+				stdout, _, err = pod.ExecCommand(clients, createdPod, "ethtool", "-i", "net2")
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:994]: ditto

File: test/conformance/tests/sriov_operator.go [hunk 7]
```diff
@@ -938,6 +938,66 @@ var _ = Describe("[sriov] operator", func() {
...
+				resourceName := "sriovnic"
+				sriovNetworkName := "sriovnetwork"
+				testNode := sriovInfos.Nodes[0]
+
+				sriovDevice, err := sriovInfos.FindOneSriovDevice(testNode)
+
+				Expect(err).ToNot(HaveOccurred())
+				createSriovPolicy(sriovDevice.Name, testNode, 5, resourceName)
+
+				ipam := `{"type": "host-local","ranges": [[{"subnet": "1.1.1.0/24"}]],"dataDir": "/run/my-orchestrator/container-ipam-state"}`
```
Comment (SchSeba) [inline test/conformance/tests/sriov_operator.go:964]: Can you please add here a range
```
"rangeStart": "10.1.1.10",
"rangeEnd": "10.1.1.20",
```

In a follow PR I will like to see a connection testing between the VF and the macvlan interface

**Pull request merged**